### ISSUE TITLE
Upstream will be allowed to update with breaking changes due to ESLint v9

### ISSUE THREAD

**Issue by MichaelDeBoey** [un-ts/eslint-plugin-import-x#170]
As @ljharb mentioned [on Discord](https://discord.com/channels/1227627367204257852/1297095313277980675/1297686122666790964), that he's going to do a once-in-a-decade breaking change now that ESLint v9 and flat config are shipped.
He also [confirmed](https://discord.com/channels/1227627367204257852/1297095313277980675/1299156957374054472) that this would also be for `eslint-plugin-import`.

Don't know what that means for this plugin, but I bet this could mean that we could at least rebase our fork to a newer version 🤔

**Comment by MichaelDeBoey** [comment]
FYI I already opened up the following PRs upstream:

- https://github.com/import-js/eslint-plugin-import/pull/3091
- https://github.com/import-js/eslint-plugin-import/pull/3092
- https://github.com/import-js/eslint-plugin-import/pull/3093

**Comment by SukkaW** [comment]
> Don't know what that means for this plugin, but I bet this could mean that we could at least rebase our fork to a newer version 🤔

We don't have to, and also impossible already.

We have completely rewritten our project in TypeScript. We have already embraced utilities from the `@typescript-eslint` project, we have already migrated all toolchains (test framework to jest, compiler from babel to tsc), and we have already been targeting Node.js 18 for months, and we are planning to introduce new APIs and new resolver design.

In short, we have already completely deviated from `eslint-plugin-import`.

**Comment by SukkaW** [comment]
> @SukkaW If I'm not mistaken, @ljharb is open to PRs to upgrade toolchains + new APIs as well

Nah, he won't agree that `eslint-plugin-import` to be rewritten in TypeScript in any foreseeable future.

**Comment by ljharb** [comment]
it must be fun to be so aggressively wrong all the time :-p

**Comment by SukkaW** [comment]
> is open to PRs to upgrade toolchains + new APIs as well

- https://github.com/import-js/eslint-plugin-import/issues/1479
- https://github.com/import-js/eslint-plugin-import/issues/2108
- https://github.com/import-js/eslint-plugin-import/issues/2111
- https://github.com/import-js/eslint-plugin-import/issues/2717

All those proposals remain opened and none of them are resolved in `eslint-plugin-import`, that's the reason behind the birth of `eslint-plugin-import-x`. 

> but I bet this could mean that we could at least rebase our fork to a newer version 🤔

When I said "we can't anyway", I mean the `eslint-plugin-import-x` has changed the entire codebase since the fork. Here is the diff:

https://github.com/import-js/eslint-plugin-import/compare/main...un-ts:eslint-plugin-import-x:master

If you are willing to resolve more than thousands of git conflictions and help us rebase, please do, I am more than happy to accept PRs from you!

> it must be fun to be so aggressively wrong all the time :-p

You didn't even comment on @MichaelDeBoey's mentioned PRs with "LGTM, waiting for the major release schedule". And all his PR towards `eslint-plugin-jsx-a11y` remains unreleased.

And also there is https://github.com/es-tooling/eslint-plugin-jsx-a11y-x/issues/1#issuecomment-2446015725, you aren't really dropping the legacy Node.js version anyway.

**Comment by ljharb** [comment]
I didn't have to, because I've communicated that to them via other means.

I'm not saying I'm going to change all of my hard-earned principles, I'm just saying that any time you're making a claim about what someone else will or will not do, you're going to automatically be wrong, because the only person you can say that about with authority is yourself.

**Comment by soryy708** [comment]
> As [@ljharb](https://github.com/ljharb) mentioned [on Discord](https://discord.com/channels/1227627367204257852/1297095313277980675/1297686122666790964), that he's going to do a once-in-a-decade breaking change now that ESLint v9 and flat config are shipped. He also [confirmed](https://discord.com/channels/1227627367204257852/1297095313277980675/1299156957374054472) that this would also be for `eslint-plugin-import`.
> 
> Don't know what that means for this plugin, but I bet this could mean that we could at least rebase our fork to a newer version 🤔

Discord links are broken. What community was that in?

---

Please clarify, what exactly are the breaking changes discussed here?

Would they invalidate https://github.com/import-js/eslint-plugin-import/issues/2053 as obsolete?

**Comment by MichaelDeBoey** [comment]
@soryy708 The invite link to the `e18e` Discord server is https://discord.gg/q8zUG7hkMe

Breaking changes PRs discussed in that thread are https://github.com/import-js/eslint-plugin-import/pull/3091 & https://github.com/import-js/eslint-plugin-import/pull/3092

**Comment by JounQin** [comment]
It still didn't happen, closing. 😂
